### DISCUSSION TITLE
How to change style/appearance of folded nodes?

### DISCUSSION THREAD

**Discussion by psmerk** [freeplane/freeplane#1102] [Help] [answered]
Dear all, I'd like to try Freeplane, but (comming from FreeMind :-), I really need a much clearer distinction between folded and unfolded nodes, as that small bubbles next to the nodes are almost "invisible" to me (and larger bubbles would take up too much space). In FreeMind, I'm able to use Tools/Preferences.../Defaults/Standard Node Style = combined, which means that folded nodes are displayed in a bubble (circled, similarly to Display selected nodes in bubbles setting in Freeplane). How can I set something like this in Freeplane? Thanks in advance.

**Comment by macmarrum** [comment]
Hi @psmerk 

I'd use Map Conditional Styles to apply a custom style to folded nodes.
The style can use a Cloud, to add an extra bubble around folded nodes.
This Map Conditional Style needs to use a Script Filter as the condition: `node.isFolded()`

I tried it and it sort-of works: [bug-isFolded.mm.zip](https://github.com/freeplane/freeplane/files/11275317/bug-isFolded.mm.zip)
There is an issue: `node.isFolded()` is only refreshed when the node text is changed or its child is moved (e.g. repositioned), deleted or added.

I've created a bug report
* https://github.com/freeplane/freeplane/issues/1103

**Comment by dpolivaev** [comment]
Isn't it available?

<img width="305" alt="combined" src="https://user-images.githubusercontent.com/1558052/233144015-d2d0d8c1-78fe-4bc6-90f0-a5d68f52e945.png">

**Reply by EdoFro** [reply]
wow!!! that's also new to me!!! 🤯

**Reply by psmerk** [reply]
> Isn't it available?
> 
> <img alt="combined" width="305" src="https://user-images.githubusercontent.com/1558052/233144015-d2d0d8c1-78fe-4bc6-90f0-a5d68f52e945.png">

Well, I'm afraid, perhaps I'm becoming blind --- please, could you be a bit more specific where I can find it? :-) I admit I have started with the last stable portable version (1.10.6) with some freemind .mm, so I can imagine it could behave somewhat differently, but now I have also tried both the "regular" stable version and the last prerelease version (1.11.1 pre29) on a new .mm (always deleting the settings etc. directory), all with Java 17 --- and it's still the same: I cannot find it.

Moreover, it seems to me like we're talking about different applications :-) --- no "Look and feel" option produces that blue arrows from your snippet, and more than that, that curved arrow on the right side looks like it allows to return to the default value, but if I change some setting, no such arrow appears (see further Antialias => Antialias edges).

Excuse the long post, but I'm completely confused. Here are screenshots of the first four tabs of my (default) preferences. Perhaps if some non-default setting adds some additional options that I don't see with default settings?

![Environment](https://user-images.githubusercontent.com/5860817/233175450-b5d9da7c-7737-4688-887e-55df74220960.png)
![Appearance](https://user-images.githubusercontent.com/5860817/233175538-b39daeb8-94de-4815-a5b1-587a68b36854.png)
![Defaults](https://user-images.githubusercontent.com/5860817/233175555-93401b82-b9a9-407b-9887-0fa824f69b2f.png)
![Behaviour](https://user-images.githubusercontent.com/5860817/233175560-d7dc040c-1e18-4340-abd9-1172cd2a0803.png)

**Reply by dpolivaev** [reply]
Some look and feels are OS-specific, so mine is only available on macOS.

The option should be available at all mentioned Freeplane versions

<img width="582" alt="nodeShape" src="https://user-images.githubusercontent.com/1558052/233196023-419eab6d-9ec6-4273-86fe-d0694b7fba73.png">

**Reply by macmarrum** [reply, marked as answer]
Fork/bubble/combined is a formatting property of a node.
In Freemind, you'd find it in the menu `Format->Change format...`.
In Freeplane, it's located in `View->Controls->Tool panel`. The easiest way to get there is right-click on the mindmap backgroud and enable the checkbox next to **Tool panel**.

But you probably want all your nodes to have **Node shape**: `Combined`. To do this in Freeplane, you need to change the **Default** node style. This can be done in `Format->Manage Styles->Edit styles`.
See also https://docs.freeplane.org/user-documentation/styles.html

**Reply by psmerk** [reply]
> In Freeplane, it's located in `View->Controls->Tool panel`. The easiest way to get there is right-click on the mindmap backgroud and enable the checkbox next to **Tool panel**.
> 
> But you probably want all your nodes to have **Node shape**: `Combined`. To do this in Freeplane, you need to change the **Default** node style. This can be done in `Format->Manage Styles->Edit styles`. See also https://docs.freeplane.org/user-documentation/styles.html

Thank you! :-) At first I was looking for it somewhere in Properties, where it's in FreeMind, but, of course, I went through the complete menu then --- and this particular option `Combined` is missing from the options in `Format->Node Core->Node Shape`, which is rather confusing: I would not expect richer set of options for the same choice hidden somewhere in one of (nondefault!) toolbars. :-)

But yes, the optimal solution to my query seems to be editing the Default style.

One more small note on this:

> Fork/bubble/combined is a formatting property of a node.
> In Freemind, you'd find it in the menu `Format->Change format...`.

As you mentioned, I need to set it globally; this is only for a given node. Moreover, in FreeMind it behaves rather strangely: I can set `Combined` with it and the particular node will be displayed correctly, but if I look at this setting for a given node later, there will be always either `Fork` or `Bubble` only. At least it will reflect the actual state of the node --- in Freeplane, the analogous(?) setting is even more confusing: as there is no `Combined` in that list of options, the same behaviour would not be surprising, but it seems to mark ("preselect" with that small blue dot) the last selected value (ever, for any node), which doesn't seem to make much sense --- or at least it's a bit counterintuitive, as in such situations, the current state is marked/preselected not only by almost all other software, but even, e.g., by the neigbouring settings `Format->Node Core->Edge Properties`. :-)
